### PR TITLE
[MAJOR] Implemented backend for call logs.

### LINKED ISSUE #152: Implement backend to add a call log
Epic/User story #130

### LINKED ISSUE #154: Implement backend to edit call log
Epic/User story #130

### LINKED ISSUE #155: Implement backend call to delete a call log
Epic/User story #130

### PR DESCRIPTION
**Pull request by huangs08**
closes #152 , closes #154 , closes #155 

- Create a call log
- Search call logs by name, email, phone number, etc.
- Edit a call log
- Delete a call log

Search function exists however it was not in the requirements. It could prove to be useful in the future, as an enhancement if needed.

### CODE REVIEW

File: backend/src/main/routes/CallRoute.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import { Roles } from '../security/Roles';
+
+@injectable()
+export default class CallRoute extends CommonRoutesConfig {
+  constructor(@inject('express-app') app: express.Application, private callService: CallService) {
+    super(app, 'AccountRoute');
+  }
+
+  configureRoutes(): express.Application {
+    this.getApp()
```
Comment (Mnx458) [inline backend/src/main/routes/CallRoute.ts:16]: I think you can combine the routes for this file under call and calls. No need to call this.get() everytime

Comment (huangs08) [reply backend/src/main/routes/CallRoute.ts:16]: You can only have one route for every this.getApp()

File: backend/src/main/services/CallService.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import { injectable } from 'tsyringe';
+import debug from 'debug';
+import { CallCreationDTO, CallUpdateDTO } from '../dto/CallDTOs';
+import HttpException from '../exceptions/HttpException';
+import { StatusCodes } from 'http-status-codes';
+import CallRepository from '../repositories/CallRepository';
+import AccountRepository from '../repositories/AccountRepository';
+import { Call } from '../models/Call';
+const log: debug.IDebugger = debug('app:userService-example');
```
Comment (nasaku898) [inline backend/src/main/services/CallService.ts:9]: Update the debug name.

**Review by nasaku898 — CHANGES_REQUESTED** [review]
Overall looks good, I added a small comment you should look at. And it seems that there is a merge conflict also.

File: backend/src/main/services/CallService.ts (hunk 2 above)
Comment (huangs08) [reply backend/src/main/services/CallService.ts]: Good catch, thanks. Updated.

**Review by nasaku898 — APPROVED** [review]

**Review by Mnx458 — APPROVED** [review]
ok good

**Pull request merged**